File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int index = 0;
        int len = 0;

        for(int i = 0; i < s.size(); ++i) {
            int l = i, r = i;
            while(l >= 0 && r < s.size() && s[l] == s[r]) {
                if (r - l + 1 > len) {
                    index = l;
                    len = r - l + 1;
                }
                l--; r++;
            }

            l = i, r = i+1;
            while(l >= 0 && r < s.size() && s[l] == s[r]) {
                if (r - l + 1 > len) {
                    index = l;
                    len = r - l + 1;
                }
                l--; r++;
            }
        }

        return s.substr(index, len);
    }
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        index = 0
        length = 0

        for i in range(len(s)):
            l = i
            r = i
            while l >= 0 and r < len(s) and s[l] == s[r]:
                if r - l + 1 > length:
                    index = l
                    length = r - l + 1
                l -= 1
                r += 1
            
            l = i
            r = i + 1
            while l >= 0 and r < len(s) and s[l] == s[r]:
                if r - l + 1 > length:
                    index = l
                    length = r - l + 1
                l -= 1
                r += 1
        
        return s[index: index + length]